webapp-django/crashstats/crashstats/static/crashstats/js/socorro/oauth2.js

```diff
@@ -110,8 +121,18 @@ var OAuth2 = (function() {
                         $.post(url, data)
                         .done(function(response) {
                             // It worked!
-                            // TODO: https://bugzilla.mozilla.org/show_bug.cgi?id=1283296
-                            document.location.reload();
+                            var next = getParameterByName('next');
+                            // only if ?next=/... exists on the current URL
+                            if (next) {
+                                // A specific URL exits.
+                                // This is most likely the case when you tried
+                                // to access a privileged URL whilst being
+                                // anonymous and being redirected.
+                                // Make sure it's on this server
+                                document.location.pathname = next;
+                            } else {
```

Comment:
`searchParams` isn't a mapping, is an object with methods you use to modify and read the parameters.

`searchParams.get` works in both Firefox and Chrome (I tested Dev Edition and whatever the latest Chrome my machine has): http://codepen.io/anon/pen/RGAAQq?editors=0012

```js
var url = new URL('https://www.mozilla.org/foo/bar?param1=4&param2=7');
console.log(url.searchParams.get('param1'));
```